addons/binding/org.openhab.binding.mailserver/src/main/java/org/openhab/binding/mailserver/internal/SmtpReceivedMessageHandler.java

```diff
@@ -0,0 +1,112 @@
+import org.openhab.binding.mailserver.handler.MailServerBridgeHandler;
+import org.openhab.binding.mailserver.handler.MailServerHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.subethamail.smtp.MessageContext;
+import org.subethamail.smtp.MessageHandler;
+import org.subethamail.smtp.RejectException;
+
+/**
+ * The {@link SmtpReceivedMessageHandler} is responsible for handling received messages
+ *
+ * @author Jereme Guenther - Initial contribution
+ */
+public class SmtpReceivedMessageHandler implements MessageHandler {
+    MessageContext ctx = null;
+
+    /**
+     * The Thing that spawned this factory instance.
+     */
+    public MailServerBridgeHandler h = null;
+
+    private Logger logger = LoggerFactory.getLogger(MailServerHandler.class);
+    protected ItemRegistry itemRegistry = null;
+    protected EventPublisher eventPublisher = null;
+
+    private String toAddress = "";
+
+    /**
+     * Constructor
+     *
+     * @param ctx the the context this incoming message is in
+     * @param h the bridge that owns this server and wants to receive the message
+     */
+    public SmtpReceivedMessageHandler(MessageContext ctx, MailServerBridgeHandler h) {
+        this.ctx = ctx;
+        this.h = h;
+    }
+
+    /**
+     * A validation function giving the chance to reject a form address
+     *
+     * @param from the incoming from address
+     */
+    @Override
+    public void from(String from) throws RejectException {
+        logger.debug("SMTP Server FROM: {}", from);
+    }
+
+    /**
+     * A validation function giving the chance to reject a recipient address
+     *
+     * @param recipient the destination address
+     */
+    @Override
+    public void recipient(String recipient) throws RejectException {
+        logger.debug("SMTP Server RECIPIENT: {}", recipient);
+    }
+
+    /**
+     * The actual receiving function that gets all the message data
+     *
+     * @param data incoming message data
+     */
+    @Override
+    public void data(InputStream data) throws IOException {
+        String convertData = this.convertStreamToString(data);
+        logger.debug("MAIL DATA: {}", convertData);
+
+        h.distributeRawMessageBodyData(toAddress, convertData);
+    }
+
+    /**
+     * A disposal method to clean up things, or to know that things are finished
+     */
+    @Override
+    public void done() {
+        logger.debug("SMTP Server Finished");
+    }
+
+    private String convertStreamToString(InputStream is) {
+        BufferedReader reader = new BufferedReader(new InputStreamReader(is));
+        StringBuilder sb = new StringBuilder();
+
+        Boolean found = false;
+        String line = null;
+        try {
+            while ((line = reader.readLine()) != null) {
+                if (!found) {
+                    String testLine = line.toLowerCase().trim();
+                    if (testLine.startsWith("To: ") || testLine.startsWith("for ")) {
```

Comment:
Maybe you should add some documentation to make it more clear why you search for *'for '*, I just checked my mail headers and found this:

```
Delivered-To: martinvw@***
Received: by 10.140.101.170 with SMTP id u39csp1634293qge;
        Tue, 11 Apr 2017 01:40:55 -0700 (PDT)
X-Received: by 10.107.27.207 with SMTP id b198mr53882862iob.218.1491900055321;
        Tue, 11 Apr 2017 01:40:55 -0700 (PDT)
Return-Path: <0100015b5c2c3b5c-eeb5777a-0ca6-4bb6-8732-77f78490233a-000000@mailer.netflix.com>
Received: from a14-191.smtp-out.amazonses.com (a14-191.smtp-out.amazonses.com. [54.240.14.191])
        by mx.google.com with ESMTPS id v8si1230712iti.101.2017.04.11.01.40.55
        for <martinvw@***>
        (version=TLS1 cipher=ECDHE-RSA-AES128-SHA bits=128/128);
        Tue, 11 Apr 2017 01:40:55 -0700 (PDT)
Received-SPF: pass (google.com: domain of 0100015b5c2c3b5c-eeb5777a-0ca6-4bb6-8732-77f78490233a-000000@mailer.netflix.com designates 54.240.14.191 as permitted sender) client-ip=54.240.14.191;
Authentication-Results: mx.google.com;
       dkim=pass header.i=@netflix.com;
       dkim=pass header.i=@amazonses.com;
       spf=pass (google.com: domain of 0100015b5c2c3b5c-eeb5777a-0ca6-4bb6-8732-77f78490233a-000000@mailer.netflix.com designates 54.240.14.191 as permitted sender) smtp.mailfrom=0100015b5c2c3b5c-eeb5777a-0ca6-4bb6-8732-77f78490233a-000000@mailer.netflix.com;
       dmarc=pass (p=REJECT sp=REJECT dis=NONE) header.from=netflix.com
Date: Tue, 11 Apr 2017 08:40:54 +0000
From: Netflix <info@mailer.netflix.com>
Sender: Netflix <info@mailer.netflix.com>
To: martinvw@***
Message-ID: <0100015b5c2c3b5c-eeb5777a-0ca6-4bb6-8732-77f78490233a-000000@email.amazonses.com>
Subject: Martin, Better Call Saul Seizoen 3 staat nu op Netflix
MIME-Version: 1.0
Content-Type: multipart/alternative; boundary="----=_Part_133227_457295461.1491900054261"
X-To: martinvw@***
Precedence: bulk
X-AppInfo: Netflix Mercury s.bdd67da.2198
X-ProcEnc: BQAtAAEBEPQBoqBseMonaKGvwYUZ34cgQp4nRkqx+SeeEqaLUNH1rlj7JPXOQvGpJs2/7m2vDIs=
X-MailingID: mercury::LR::12182::80E26024DA606C382E16C59E5D28962F64523AA4::<15B5C2C3AF6@netflix.com>
X-GroupId: 4
X-localeCountry: nl::NL
X-ProcInfo: mercuryfigment::mercuryfigment-i-0edcf2d0d26f578c8::eventconsumer::eventconsumer-figment-i-04132d8627bb807fb::mercurymp-daily-i-0ba3920bb965bde24::00:23:02:56.426
X-EvCacheKey: 12182.EMAIL.EMAIL_NEW_SEASON_ALERT.-0005.-5.nl.NL
X-Version: 2521
X-SES-Outgoing: 2017.04.11-54.240.14.191
Feedback-ID: 1.us-east-1.TW2kQXBPKzEfHzcyZOUFPUd4i/tuZQjQCCDIsBA/Brk=:AmazonSES
```

Some questions:

- are you okay what also receiving the <> around it?
- Are you aware that the 'To' will never match since you converted the input to lowercase
- do you want to support cc or do you already do that because you filter out the 'for' header